### ISSUE TITLE
ACME won't generate additional SAN Certificate

### ISSUE THREAD

**Issue by fabianwloch** [mailcow/mailcow-dockerized#3527]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [X] I understand, that not following or deleting the below instructions, will result in immediate closing and deletion of my issue.
- [X] I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

**Description of the bug**:

I recently installed Mailcow (last week), which worked fine. Today, I noticed that I might want to use another domain for some E-Mail-Adresses I use, so I ordered a new domain, set up the DNS records, waited a day and now tried to configure it in mailcow.
nginx works as expected, but the certificated that gets served by nginx only contains the old/original domain, not the new one that I try to configure.
My mailcow.conf:
```
MAILCOW_HOSTNAME=mail.DOMAIN1.is
ADDITIONAL_SAN=DOMAIN1.is,DOMAIN2.org,mail.DOMAIN2.org
```

**Docker container logs of affected containers**:

```
root@mail:/opt/mailcow-dockerized# docker-compose logs --tail=200 -f acme-mailcow
Attaching to mailcowdockerized_acme-mailcow_1
acme-mailcow_1       | Sat May  9 10:30:19 CEST 2020 - Waiting for Docker API...OK
acme-mailcow_1       | Sat May  9 10:30:19 CEST 2020 - Waiting for database... Uptime: 8  Threads: 28  Questions: 260  Slow queries: 0  Opens: 120  Flush tables: 1  Open tables: 112  Queries per second avg: 32.500
acme-mailcow_1       | OK
acme-mailcow_1       | Sat May  9 10:30:19 CEST 2020 - Waiting for Nginx... OK
acme-mailcow_1       | Sat May  9 10:30:26 CEST 2020 - Waiting for domain table... OK
acme-mailcow_1       | Sat May  9 10:30:26 CEST 2020 - Initializing, please wait... 
acme-mailcow_1       | Sat May  9 10:30:26 CEST 2020 - Using existing domain rsa key /var/lib/acme/acme/key.pem
acme-mailcow_1       | Sat May  9 10:30:26 CEST 2020 - Using existing Lets Encrypt account key /var/lib/acme/acme/account.pem
acme-mailcow_1       | Sat May  9 10:30:26 CEST 2020 - Detecting IP addresses... Sat May  9 10:31:03 CEST 2020 - Waiting for Docker API...OK
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Waiting for database... Uptime: 52  Threads: 33  Questions: 348  Slow queries: 0  Opens: 123  Flush tables: 1  Open tables: 115  Queries per second avg: 6.692
acme-mailcow_1       | OK
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Waiting for Nginx... OK
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Waiting for domain table... OK
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Initializing, please wait... 
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Using existing domain rsa key /var/lib/acme/acme/key.pem
acme-mailcow_1       | Sat May  9 10:31:03 CEST 2020 - Using existing Lets Encrypt account key /var/lib/acme/acme/account.pem

```

**Reproduction of said bug**:

I of course restarted the acme container. When this didn't work, I restarted the whole mailcow installation. When this didn't work, I tried the "force renewal" from the installation guide.
I noticed the hint "The file will be deleted automatically." for the "data/assets/ssl/force_renew" file, and that it did not get deleted. After several tries of restarting the container and the whole installation through "docker-compose down / pull / up -d", I restarted the whole server, sadly this didn't help. So at this point, I gave up and created this Issue.

**System information**:

| Question | Answer |
| --- | --- |
| My operating system | Debian Buster 10.3 |
| Is Apparmor, SELinux or similar active? | apparmor in it's default config from Debian standard installation, didn't changed anything here |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | Dedicated Hardware (i5 2400, 4GB DDR3, 500GB HDD) |
| Docker Version (`docker version`) | 19.03.8 |
| Docker-Compose Version (`docker-compose version`) | 1.25.5, build 8a1c60f6 |
| Reverse proxy (custom solution) | Default mailcow configuration, didn't installed anything additional on the server |

- Output of `git diff origin/master`, any other changes to the code? If so, **please post them**.
Only stronger SSL ciphers:
/data/conf/nginx/includes/site-defaults.conf
```
-ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+ssl_ciphers EECDH+AESGCM:EDH+AESGCM;
```

- All third-party firewalls and custom iptables rules are unsupported. *Please check the Docker docs about how to use Docker with your own ruleset*. Nevertheless, iptabels output can help us to help you: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat`.

I indeed use a firewall regularly, when I read this part right here I flushed all rules and tried again - didn't help.

- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output.

DNS works fine

**Comment by fabianwloch** [comment]
Alright. Sorry for creating the Issue, I guess I found the problem.
I configured IPv6 on the server, and by now, after like 30 tries, I finally get an error message in the acme container log: It can't resolve the IPv6 Address from the DNS, and I can't reach it from outside. It is correctly configured on my server tho, so I guess I'll ask my hoster.
I write again if this was indeed the problem, but it seems like it. Just weird that I didn't got this error like 3 hours before... :(

**Comment by andryyy** [comment]
Please do always post the full logs, don't remove things, you think are not important. :(

This is not a bug, but a network question, the forums would be better suited for this. :)

Can you connect to the outside world via IPv6? Like `docker-compose exec acme-mailcow ping6 google.com` or `docker-compose exec acme-mailcow curl ip6.mailcow.email`?

**Comment by fabianwloch** [comment]
That actually was the full log while testing this for multiple hours - that's why I created the issue in the first place. I didn't found a hint for 3 hours, and I don't know why.
But after this time, and after creating the issue, there was randomly new output, which lead me to my second answer.
It now works, and I guess I found the problem: The hoster of my server replied me, that they had/have some networking issues.
